### PR TITLE
Add PusheratorBuild chaining API

### PR DESCRIPTION
**Pull request by MingweiSamuel**
This adds an iterator-style functional-chaining builder API which returns a `Pivot` instance.

Next I would like to actually integrate or add a layer on top of this so the scheduled-layer APIs and this can work seamlessly. Right now this is still used separately within the scheduled layer.

Depends on #26

### CODE REVIEW

File: covid_tracing/src/main.rs [hunk 1]
```diff
@@ -5,8 +5,8 @@ use std::time::Duration;
 
 use hydroflow::compiled::pull::JoinState;
 use hydroflow::compiled::pull::SymmetricHashJoin;
-use hydroflow::compiled::{ForEach, Pivot, Tee};
-use hydroflow::scheduled::collections::Iter;
+use hydroflow::compiled::{InputBuild, IteratorToPusherator, PusheratorBuild}; //{ForEach, Pivot, Tee};
```
Comment (justinj) [inline covid_tracing/src/main.rs:8]: nit: remove comment

File: covid_tracing/src/main.rs [hunk 2]
```diff
@@ -62,28 +62,26 @@ fn main() {
...
-                let notif_push = ForEach::new(|exposed_person: (Pid, DateTime)| {
-                    // println!("DEBUG: will_notif {} {}", exposed_person.0, exposed_person.1);
-                    notifs_send.give(Some(exposed_person));
-                });
-                let loop_push = ForEach::new(|exposed_person: (Pid, DateTime)| {
-                    // println!("DEBUG: will_loop {}, {}", exposed_person.0, exposed_person.1);
-                    loop_send.give(Some(exposed_person));
-                });
-                let push_exposed = Tee::new(notif_push, loop_push);
+                let pivot = new_exposed
```
Comment (justinj) [inline covid_tracing/src/main.rs:72]: way better 👍

File: hydroflow/src/lang/lattice/dom_pair.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
+use std::cmp::Ordering;
```
Comment (justinj) [inline hydroflow/src/lang/lattice/dom_pair.rs:1]: dom pair kind of sketches me out, can we add a big disclaimer that you have to be very careful in order for this to be a lattice?

File: test.txt [hunk 4]
```diff
@@ -0,0 +1,95 @@
+
```
Comment (justinj) [inline test.txt:1]: nit: this file added accidentally? if you were trying to regenerate the contents you have to set the env var `REWRITE`.

**Review by justinj — APPROVED** [review]
LGTM! The solution with the `Build` part of each operator was tidier than I expected, seems pretty nice.

File: covid_tracing/src/main.rs (hunk 1 above)
Comment (MingweiSamuel) [reply covid_tracing/src/main.rs:8]: 🙏

**Pull request merged**